### PR TITLE
bpo-23378: Add an extend action to argparse

### PR DESCRIPTION
**Pull request by isidentical**
Add an extend action to argparse

https://bugs.python.org/issue23378

### CODE REVIEW

**Review by asottile — COMMENTED** [review]
patch looks good, are there other `nargs=...` that this makes sense with? should it produce an error in the nonsensical cases? should it default to `nargs='+'`?

**Comment by isidentical** [comment]
`nargs=N` doesn't make any sense (it doesn't need `extend` but `extend` can be used like `append`).  but i dont think we should produce a `ValueError` etc.

**Review by furkanonder — APPROVED** [review]
Nice fix.

File: Doc/library/argparse.rst [hunk 1]
```diff
@@ -797,6 +797,15 @@ how the command-line arguments should be handled. The supplied actions are:
     >>> parser.parse_args(['--version'])
     PROG 2.0
 
+* ``'extend'`` - This stores a list, and extends each argument value to the
+  list.
+  Example usage::
+
+    >>> parser = argparse.ArgumentParser()
+    >>> parser.add_argument("--foo", action="extend", nargs="+", type=str)
+    >>> parser.parse_args("--foo f1 --foo f2 f3 f4".split())
```
Comment (gvanrossum) [inline Doc/library/argparse.rst]: I would remove the `split()` call and just write `parser.parse_args("--foo",  "f1", "--foo", "f2", "f3", " f4")`

Comment (isidentical) [reply Doc/library/argparse.rst]: resolved

**Review by gvanrossum — APPROVED** [review]
Yup!

**Pull request merged**